### ISSUE TITLE
'geckodriver' executable needs to be in PATH

### ISSUE THREAD

**Issue by tXambe** [sundowndev/PhoneInfoga#84]
Hello,

When execute PhoneInfoga I got this error:

[+] Scan URL: https://www.411.com/phone/3-666-666-666
Would you like to use an additional format for this number ? (y/N) N
[i] ---- Web pages footprints ----
[i] Searching for footprints on web pages... (limit=10)
Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/selenium/webdriver/common/service.py", line 76, in start
    stdin=PIPE)
  File "/usr/lib/python3.7/subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.7/subprocess.py", line 1522, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'geckodriver': 'geckodriver'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "phoneinfoga.py", line 95, in <module>
    main()
  File "phoneinfoga.py", line 71, in main
    scanNumber(args.number)
  File "phoneinfoga.py", line 38, in scanNumber
    osintScan(number)
  File "/opt/OSINT/PhoneInfoga/scanners/footprints.py", line 173, in osintScan
    for result in search(req, stop=10):
  File "/opt/OSINT/PhoneInfoga/lib/googlesearch.py", line 35, in search
    browser = webdriver.Firefox()
  File "/root/.local/lib/python3.7/site-packages/selenium/webdriver/firefox/webdriver.py", line 164, in __init__
    self.service.start()
  File "/root/.local/lib/python3.7/site-packages/selenium/webdriver/common/service.py", line 83, in start
    os.path.basename(self.path), self.start_error_message)
selenium.common.exceptions.WebDriverException: Message: 'geckodriver' executable needs to be in PATH. 

A greeting and thanks

**Comment by sundowndev** [comment]
It seems you didn't installed geckodriver properly. See [this comment on stackoverflow](https://stackoverflow.com/a/40437690).

1. Are you using Docker ?
2. What command did you run ?
3. What is your OS ?
4. Do you have the latest version of Firefox installed ?

**Comment by C8be** [comment]
I have the same Problem and
1. I don't use Docker
2. I ran the command python3 phoneinfoga.py -n %Number%
3. Ubuntu 16.04 LTS
4. Yes I updated Firefox
5 and it says geckodriver: command not found

**Comment by mashsajid** [comment]
wget https://github.com/mozilla/geckodriver/releases/download/v0.23.0/geckodriver-v0.23.0-linux64.tar.gz
sudo sh -c 'tar -x geckodriver -zf geckodriver-v0.23.0-linux64.tar.gz -O > /usr/bin/geckodriver'
sudo chmod +x /usr/bin/geckodriver
rm geckodriver-v0.23.0-linux64.tar.gz

**Comment by sundowndev** [comment]
Feel free to add further details, considering this as resolved.

**Comment by popduck07** [comment]
hey sundown, I'm relatively new to all this and I read the previous thread. I'm having the same issue, can you dumb it down for me? I'm using kali linux and I ran the command: "python3 phoneinfoga.py -n "(+X) XXXXXXXXXX".  After I had responded with 'n' when it asked me about custom format it outputted:

 Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/selenium/webdriver/common/service.py", line 76, in start
    stdin=PIPE)
  File "/usr/lib/python3.7/subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.7/subprocess.py", line 1522, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'geckodriver': 'geckodriver'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "phoneinfoga.py", line 95, in <module>
    main()
  File "phoneinfoga.py", line 71, in main
    scanNumber(args.number)
  File "phoneinfoga.py", line 38, in scanNumber
    osintScan(number)
  File "/root/PhoneInfoga/scanners/footprints.py", line 173, in osintScan
    for result in search(req, stop=10):
  File "/root/PhoneInfoga/lib/googlesearch.py", line 35, in search
    browser = webdriver.Firefox()
  File "/usr/local/lib/python3.7/dist-packages/selenium/webdriver/firefox/webdriver.py", line 164, in __init__
    self.service.start()
  File "/usr/local/lib/python3.7/dist-packages/selenium/webdriver/common/service.py", line 83, in start
    os.path.basename(self.path), self.start_error_message)
selenium.common.exceptions.WebDriverException: Message: 'geckodriver' executable needs to be in PATH. 

how do I properly install the geckodriver? when I typed "pip install geckodriver" it outputted 

Collecting geckodriver
Could not install packages due to an EnvironmentError: 404 Client Error: Not Found for url: https://pypi.org/simple/geckodriver/

I would appreciate some help

**Comment by sundowndev** [comment]
@popduck07 https://sundowndev.github.io/PhoneInfoga/install/#install-the-geckodriver

**Comment by RaviFefar** [comment]
The problem will fix only Linux users

1. Download the **geckodriver** from `wget https://github.com/mozilla/geckodriver/releases/download/v0.23.0/geckodriver-v0.23.0-linux64.tar.gz`

2. **Extract** and **unzip** and **move** the geckodriver file to `/usr/local/bin/` directory.
3. `sudo chmod +x /usr/local/bin/geckodriver`
3. `export PATH=$PATH:/usr/local/bin/geckodriver`
4. Now Run python program with selenium Firefox webdriver.
